core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -43,44 +50,84 @@ import scala.concurrent.Future
  *                                     +----------------------+
  */
 object Scheduler {
-  case class MesosConnection(mesosHostName: String)
+  type SpecInput = (SpecsSnapshot, Source[SpecUpdated, Any])
 
-  def connect(mesosHostName: String): Future[(MesosConnection, Flow[SpecEvent, StateEvent, NotUsed])] = {
-    val flow = Flow.fromGraph {
-      GraphDSL.create(Scheduler.unconnectedGraph, FakeMesos.flow)((_, _) => NotUsed) { implicit builder =>
-        { (graph, mockMesos) =>
+  type StateOutput = (StateSnapshot, Source[StateEvent, Any])
+
+  case class MesosConnection(mesosHostName: String, mesosCallFactory: MesosCalls, frameworkId: FrameworkID, schedulerFlow: Flow[SpecInput, StateOutput, NotUsed])
+
+  // TODO - provide abstraction for FrameworkID persistence, recovery, etc.; we'll want to exclude the frameworkId
+  // TODO - This layer should provide a cancellation mechanism and an auto-reconnect mechanism
```

Comment:
Cancellation - yes, auto-reconnect - I don't think so (especially when thinking about HA frameworks).

Reply:
No? We should crash on Mesos connection loss?

Reply:
Signal that to the orchestration layer and let it decide, I'd say

Reply:
Discussed